src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/UpdateAzureMLWebService.cs

```diff
@@ -0,0 +1,195 @@
+using Microsoft.WindowsAzure.Commands.Common;
+
+namespace Microsoft.Azure.Commands.MachineLearning
+{
+    [Cmdlet(
+        VerbsData.Update, 
+        WebServicesCmdletBase.CommandletSuffix, 
+        SupportsShouldProcess = true)]
+    [OutputType(typeof(WebService))]
+    public class UpdateAzureMLWebService : WebServicesCmdletBase
+    {
+        protected const string UpdateFromArgumentsParameterSet = 
+            "Update specific properties of the .";
+        protected const string UpdateFromObjectParameterSet = 
+            "Create a new Azure ML webservice from a WebService instance definition.";
+
+        [Parameter(Mandatory = true, HelpMessage = "The name of the resource group for the Azure ML web service.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The name of the web service.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New title for the web service.")]
+        public string Title { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New description for the web service.")]
+        public string Description { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "Mark the service as readonly.")]
+        public SwitchParameter IsReadOnly { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New access keys for the web service.")]
+        public WebServiceKeys Keys { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New access key for the storage account associated with the web service. This allows for key rotation if needed.")]
+        public string StorageAccountKey { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New diagnostics settings for the web service.")]
+        public DiagnosticsConfiguration Diagnostics { get; set; }
+
+        [Parameter(
+            ParameterSetName = UpdateAzureMLWebService.UpdateFromArgumentsParameterSet, 
+            Mandatory = false, 
+            HelpMessage = "New realtime endpoint runtime settings for the web service.")]
+        public RealtimeConfiguration RealtimeConfiguration { get; set; }
```

Comment:
We do have a test case shows how to do it. So I think our users should be able to do the same thing. For the definition of each structure, the user could find it out from the web service definition.

            # Update realtime endpoint settings and the service key by using in line parameters
            $newPrimaryKey = 'highly secure key'
            LogOutput "Updating in line properties on service $($svc.Id)"
            $updatedSvc2 = Update-AzureRmMlWebService -ResourceGroupName $rgName -Name $webServiceName -RealtimeConfiguration @{ MaxConcurrentCalls = 30 } -Keys @{ Primary = $newPrimaryKey }
            Assert-NotNull $updatedSvc2
            LogOutput "Update has completed."
            $update2ModifiedOn = [datetime]::Parse($updatedSvc2.Properties.ModifiedOn)
            LogOutput "Web service's last modified time stamp: $update2ModifiedOn"